packages/astro/src/vite-plugin-markdown-legacy/index.ts

```diff
@@ -0,0 +1,285 @@
+				return resolvedId?.id.replace(MARKDOWN_CONTENT_FLAG, '');
+			}
+			// If the markdown file is imported from another file via ESM, resolve a JS representation
+			// that defers the markdown -> HTML rendering until it is needed. This is especially useful
+			// when fetching and then filtering many markdown files, like with import.meta.glob() or Astro.glob().
+			// Otherwise, resolve directly to the actual component.
+			if (id.endsWith('.md') && !isRootImport(importer)) {
+				const resolvedId = await this.resolve(id, importer, { skipSelf: true, ...options });
+				if (resolvedId) {
+					return resolvedId.id + MARKDOWN_IMPORT_FLAG;
+				}
+			}
+			// In all other cases, we do nothing and rely on normal Vite resolution.
+			return undefined;
+		},
+		async load(id, opts) {
+			// A markdown file has been imported via ESM!
+			// Return the file's JS representation, including all Markdown
+			// frontmatter and a deferred `import() of the compiled markdown content.
+			if (id.endsWith(`.md${MARKDOWN_IMPORT_FLAG}`)) {
+				const { fileId, fileUrl } = getFileInfo(id, config);
+
+				const source = await fs.promises.readFile(fileId, 'utf8');
+				const { data: frontmatter, content: rawContent } = safeMatter(source, fileId);
+				return {
+					code: `
+						// Static
+						export const frontmatter = ${escapeViteEnvReferences(JSON.stringify(frontmatter))};
+						export const file = ${JSON.stringify(fileId)};
+						export const url = ${JSON.stringify(fileUrl)};
+						export function rawContent() {
+							return ${escapeViteEnvReferences(JSON.stringify(rawContent))};
+						}
+						export async function compiledContent() {
+							return load().then((m) => m.compiledContent());
+						}
+						export function $$loadMetadata() {
+							return load().then((m) => m.$$metadata);
+						}
+
+						// Deferred
+						export default async function load() {
+							return (await import(${JSON.stringify(fileId + MARKDOWN_CONTENT_FLAG)}));
+						}
+						export function Content(...args) {
+							return load().then((m) => m.default(...args));
+						}
+						Content.isAstroComponentFactory = true;
+						export function getHeadings() {
+							return load().then((m) => m.metadata.headings);
+						}
+						export function getHeaders() {
+							console.warn('getHeaders() have been deprecated. Use getHeadings() function instead.');
+							return load().then((m) => m.metadata.headings);
+						};`,
+					map: null,
+				};
+			}
+
+			// A markdown file is being rendered! This markdown file was either imported
+			// directly as a page in Vite, or it was a deferred render from a JS module.
+			// This returns the compiled markdown -> astro component that renders to HTML.
+			if (id.endsWith('.md')) {
+				const filename = normalizeFilename(id);
+				const source = await fs.promises.readFile(filename, 'utf8');
+				const renderOpts = config.markdown;
+				const isAstroFlavoredMd = config.legacy.astroFlavoredMarkdown;
```

Comment:
no longer needed! It would be great to remove these branching logic flows from the plugin code now that it's flipped on/off outside of this plugin

Reply:
Fair enough! Just left it untouched because we plan to delete this module anyways (with no future changes). I can refactor though.

Reply:
Done 👍